src/mesido/pycml/component_library/milp/gas/compressor.py

```diff
@@ -0,0 +1,30 @@
+from numpy import nan
+
+from .gas_base import GasTwoPort
+from .._internal import BaseAsset
+
+
+class Compressor(GasTwoPort, BaseAsset):
+    """
+    A gas compressor increases the pressure level of the flow.
+    """
+
+    def __init__(self, name, **modifiers):
+        super().__init__(name, **modifiers)
+
+        self.component_type = "compressor"
+        self.min_head = 30.0
+
+        self.Q_nominal_in = nan
+        self.Q_nominal_out = nan
+
+        self.density_in = 2.5e3  # H2 density [g/m3] at 30bar
+        self.density_out = 2.5e3  # H2 density [g/m3] at 30bar
+
+        self.add_equation(
+            (
+                (self.GasIn.Q * self.density_in - self.GasOut.Q * self.density_out)
+                / (self.Q_nominal_in * self.density_in * self.Q_nominal_out * self.density_out)
+                ** 0.5
+            )
+        )
```

Comment:
Don't we also want to add conservation of mass?

Reply:
This is at the moment guaranteed by the constraints in the pipe

Reply:
but the mass flows from one pipe through the compressor to the other pipe, I don't see how this is guaranteed by constraints in the pipe.

Reply:
You are talking about the mass_flow variable right?

In principle the equation above is making sure mass is conserved as it uses the density. The mass_flow variable is constrained with the density in the pipe.

Reply:
Yes I know see, the mass_flow variable is indeed linked to Q at the pipe in and outlet, so that is fine.